Dockerfile

```
@@ -0,0 +1,26 @@
+
+## Dependency Stage ##
+FROM node:18-alpine AS dep
+WORKDIR /app
+COPY package.json ./package.json
+COPY yarn.lock ./yarn.lock
+# copy `dist` if it exists already
+COPY dis[t]/web ./dist/
+COPY node_module[s] ./node_modules
+# git is needed for jinter
+RUN apk add git
+# don't rebuild if you don't have to
+RUN if [ ! -d 'node_modules' ]; then yarn ci; fi
```

```suggestion
RUN if [ ! -d 'dist/web' ]; then yarn ci; fi
```
Along with the above comment, this should also be changed, still removing the need to reinstall dependencies if the build is already complete, but not using the system deps because they could be different than those needed in the docker image.